src/common/gpio_servo.rs

```diff
@@ -154,7 +154,14 @@ impl<B> Servo for GpioServo<B>
 where
     B: Board,
 {
+    /// this implementation of move_to clamps the angle to the range determined
+    /// by min_angle_deg and max_angle_deg
     fn move_to(&mut self, angle_deg: u32) -> anyhow::Result<()> {
+        let angle_deg = match angle_deg {
```

Comment:
I think you can actually use the [clamp](https://doc.rust-lang.org/core/cmp/trait.Ord.html#method.clamp) here.